tests/test_queries.py

```diff
@@ -0,0 +1,890 @@
+                include_dsgrid_dataset_components=True,
+                dataset_ids=[
+                    "conus_2022_reference_comstock",
+                    "conus_2022_reference_resstock",
+                    # "tempo_conus_2022",
+                ],
+                # TODO: need to allow min/max/sum of metric on other supplemental dimensions
+                dimension_filters=[
+                    DimensionFilterExpressionModel(
+                        dimension_type=DimensionType.GEOGRAPHY,
+                        query_name="county",
+                        operator="==",
+                        value="06037",
+                    ),
+                ],
+                metric_reductions=[
+                    MetricReductionModel(
+                        dimension_query_name="electricity",
+                        operation=self._operation,
+                    ),
+                ],
+                # filtered_datasets=[],
+            ),
+        )
+        return self._model
+
+    def get_output_electricity_op(self, operation):
+        df = self._spark.read.parquet(str(self.output_dir / self.name / "table.parquet"))
+        value = perform_op(df, f"all_electricity_{operation.__name__}", operation)
+        return value
+
+    def validate(self):
+        county = self.get_filtered_county_id()
+        expected_base = self.get_raw_stats()["by_county"][county]["comstock_resstock"]
+        if self._operation == "sum":
+            expected = expected_base["sum"]["electricity"]
+            actual = self.get_output_electricity_op(F.sum)
+        elif self._operation == "max":
+            expected = expected_base["max"]["electricity"]
+            actual = self.get_output_electricity_op(F.max)
+        else:
+            logger.warning("Validation is not yet supported for %s", self._operation)
+            return True
+
+        result = math.isclose(expected, actual)
+        if not result:
+            ratio = actual / expected
+            logger.error(
+                "Mismatch in values. actual=%s expected=%s ratio=%s", actual, expected, ratio
+            )
+        return result
+
+
+class QueryTestElectricityValues(QueryTestBase):
+
+    NAME = "electricity-values"
+
+    def make_query(self):
+        self._model = ProjectQueryResultModel(
+            name=self.NAME,
+            project=ProjectConstraintsModel(
+                project_id="dsgrid_conus_2022",
+                include_dsgrid_dataset_components=True,
+                dataset_ids=[
+                    "conus_2022_reference_comstock",
+                    "conus_2022_reference_resstock",
+                    # "tempo_conus_2022",
+                ],
+                dimension_filters=[
+                    DimensionFilterColumnOperatorModel(
+                        dimension_type=DimensionType.METRIC,
+                        query_name="electricity",
+                        operator="like",
+                        value="%",
```

Comment:
What is this? Why %? Is this equivelent to `col like "%electricity"`? If so, does the prefix/suffix placement of the % matter here?

Reply:
This filter is one way to saw, "Filter out all columns that are not electricity end uses. Keep the end use column names. Do not perform any reduction/aggregation." The Spark SQL `like` was supposed to match any string - just `"%"`. However, this SQL like statement isn't actually used here. Instead the code filters out any dimension record ID that does not have a mapping. natural_gas_heating does not map to electricity, so it gets filtered out. I need to change this interface.

Reply:
This is now fixed. We select only the dimensions that map this supplemental dimension and then apply the filter. Since the filter value is "%", we keep all. If you wanted to keep only columns with "cooling" in the name, `value="%cooling"`. 

This solution may not be immediately intuitive for users and I'm open to a change.

Reply:
Per our meeting, it's non-intuitive that the filter is being applied on the "electricity" query name but the column selections are on the detailed end-uses. Maybe these are really two different operations/filters on two different query names? Maybe default behavior when joining a new query name is to keep the old one around unless it is aggregated out of existence? Is that possible?